tmt/steps/execute/__init__.py

```diff
@@ -55,32 +58,101 @@
 # Metadata file with details about the current test
 TEST_METADATA_FILENAME = 'metadata.yaml'
 
-# Scripts source directory
+#: Scripts source directory
 SCRIPTS_SRC_DIR = tmt.utils.resource_files('steps/execute/scripts')
 
+#: The default scripts destination directory
+SCRIPTS_DEST_DIR = Path("/var/tmp/tmt/bin")  # noqa: S108 insecure usage of temporary dir
+
 
 @dataclass
 class Script:
-    """ Represents a script provided by the internal executor """
+    """
+    Represents a script provided by the internal executor.
+
+    Must be used as a context manager. The context manager returns
+    the source file path.
+
+    The source file name matches the name of the file specified via
+    the ``path` attribute and must be located in the directory specified
+    via :py:data:`SCRIPTS_SRC_DIR` variable.
+    """
 
     path: Path
     aliases: list[Path]
     related_variables: list[str]
 
+    def __enter__(self) -> Path:
+        return SCRIPTS_SRC_DIR / self.path.name
+
+    def __exit__(self, *args: object) -> None:
+        pass
+
 
 @dataclass
 class ScriptCreatingFile(Script):
-    """ Represents a script which creates a file """
+    """
+    Represents a script which creates a file.
+
+    See :py:class:`Script` for more details.
+    """
 
     created_file: str
 
 
+@dataclass
+class ScriptTemplate(Script):
+    """
+    Represents a Jinja2 templated script.
+
+    Must be used as a context manager. The context manager returns
+    the source file path.
+
+    The source file name is constructed from the name of the file specified
+    via the ``path` attribute that is extended with the ``.j2`` suffix.
+    The template file must be located in the directory specified
+    via :py:data:`SCRIPTS_SRC_DIR` variable.
+    """
+
+    context: dict[str, str]
+
+    _rendered_script_path: Optional[Path] = None
+
+    def __enter__(self) -> Path:
+        with NamedTemporaryFile(mode='w', delete=False) as rendered_script:
+            rendered_script.write(render_template_file(
+                SCRIPTS_SRC_DIR / f"{self.path.name}.j2", None, **self.context))
+
+        self._rendered_script_path = Path(rendered_script.name)
+
+        return self._rendered_script_path
+
+    def __exit__(self, *args: object) -> None:
+        assert self._rendered_script_path
+        os.unlink(self._rendered_script_path)
+
+
+def effective_scripts_dest_dir() -> Path:
+    """
+    Find out what the actual scripts destination directory is.
+
+    If the ``TMT_SCRIPTS_DEST_DIR`` variable is set, it is used as the scripts
+    destination directory. Otherwise, the default of :py:data:`SCRIPTS_DEST_DIR`
+    is used.
```

Comment:
```suggestion
    If the ``TMT_SCRIPTS_DEST_DIR`` environment variable is set, it is used
    as the scripts destination directory. Otherwise, the default
    of :py:data:`SCRIPTS_DEST_DIR` is used.
```